reactor-core/src/main/java/reactor/core/publisher/MonoSingle.java

```diff
@@ -169,13 +166,13 @@ public void onComplete() {
 					complete(t);
 				}
 				else {
-					actual.onError(Operators.onOperatorError(this,
+					actual().onError(Operators.onOperatorError(this,
 							new NoSuchElementException("Source was empty"),
-							actual.currentContext()));
+							actual().currentContext()));
 				}
 			}
 			else if (c == 1) {
-				complete(this.value);
+				complete();
```

Comment:
I wonder if having both `complete(T)` and `complete()` isn't going to introduce more confusion. Maybe we can find better naming for both? The `complete()` method doesn't have any javadoc, which doesn't help.